Signal/src/ViewControllers/MessageComposeTableViewController.m

```diff
@@ -59,6 +60,18 @@ - (void)loadView
     _contactsViewHelper.delegate = self;
     _nonContactAccountSet = [NSMutableSet set];
 
+    ReminderView *contactsPermissionReminderView = [[ReminderView alloc]
+        initWithText:NSLocalizedString(@"COMPOSE_SCREEN_MISSING_CONTACTS_PERMISSION",
+                         @"Multiline label explaining why compose-screen contact picker is empty.")
+           tapAction:^{
+               [[UIApplication sharedApplication] openSystemSettings];
+           }];
+    [self.view addSubview:contactsPermissionReminderView];
+    [contactsPermissionReminderView autoPinWidthToSuperview];
+    [contactsPermissionReminderView autoPinEdgeToSuperviewMargin:ALEdgeTop];
+    _hideContactsPermissionReminderViewConstraint =
+        [contactsPermissionReminderView autoSetDimension:ALDimensionHeight toSize:0];
```

Comment:
I wonder if we should make this a cell in the table. That way it could scroll offscreen, which would be especially nice on smaller form factors.